.github/workflows/ci.yml

```diff
@@ -67,41 +71,58 @@ jobs:
             ${{ matrix.name }}-build
       - name: Setup
         run: |
-          # open shell once for initial setup
-          nix-shell --run true
+          # open nix-shell once for initial setup
+          true
         if: matrix.os != 'windows-latest'
       # remove problematic tests for sanitized build
       - name: Pre build
         run: rm tests/compiler/StackOverflow.lean tests/compiler/StackOverflowTask.lean
         if: matrix.name == 'Linux fsanitize'
       - name: Build
-        run: nix-shell --run ./script/ci.sh
-        if: matrix.os != 'windows-latest'
-      - name: Build
-        run: msys2do ./script/ci.sh
-        if: matrix.os == 'windows-latest'
-      - name: Test
-        run: nix-shell --run "cd build/stage0.5; ctest -j4 --output-on-failure < /dev/null"
-        if: matrix.os != 'windows-latest'
+        run: |
+          mkdir build
+          cd build
+          cmake .. ${{ matrix.CMAKE_OPTIONS }}
+          make -j4
+      - name: Patch
+        run: patchelf --set-interpreter /lib64/ld-linux-x86-64.so.2 --remove-rpath build/stage0.5/bin/lean
+        if: matrix.name == 'Linux release'
+      - name: Patch
+        run: |
+          otool -l build/stage0.5/bin/lean
+          for lib in $(otool -l build/stage0.5/bin/lean | cut -d' ' -f1); do
+            install_name_tool -change "$lib" "/usr/lib/$(basename $lib)" build/stage0.5/bin/lean
+          done
```

Comment:
Huh, this might actually work. @leodemoura Could you try the current macOS artifact again?